### PR TITLE
MULE-10302 Define and implement MuleEventBuilder API

### PR DESCRIPTION
**Pull request by dfeist**

### CODE REVIEW

**Comment by aiannucci** [comment]
jenkins validate

**Comment by aiannucci** [comment]
jenkins validate

File: core/src/main/java/org/mule/runtime/core/model/resolvers/MethodHeaderPropertyEntryPointResolver.java [hunk 1]
```diff
@@ -49,7 +50,12 @@ public InvocationResult invoke(Object component, MuleEventContext context) throw
     // MULE-4874: this is needed in order to execute the transformers before determining the methodProp
     Object[] payload = getPayloadFromMessage(context);
 
-    Object methodProp = context.getEvent().getFlowVariable(getMethodProperty());
+    Object methodProp = null;
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/model/resolvers/MethodHeaderPropertyEntryPointResolver.java]: This seems to be repeated in several places. Can it be extracted somewhere?

File: tests/integration/foo/d8b757e0-693b-11e6-8bd9-f2ee33498a70.dat [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#
+#Tue Aug 23 11:14:04 ART 2016
```
Comment (elrodro83) [inline tests/integration/foo/d8b757e0-693b-11e6-8bd9-f2ee33498a70.dat]: remove this files

File: modules/ws/src/main/java/org/mule/runtime/module/ws/consumer/WSConsumer.java [hunk 3]
```diff
@@ -210,15 +210,11 @@ protected MuleEvent processResponse(MuleEvent response, final MuleEvent request)
...
+      private Object propertyValue;
 
       @Override
       protected MuleEvent processRequest(MuleEvent event) throws MuleException {
-        try {
-          propertyValue = event.getFlowVariable(propertyName);
-        } catch (NoSuchElementException nsee) {
-          // Ignore
-        }
+        propertyValue = DefaultMuleEvent.getFlowVariableOrNull(propertyName, event);
```
Comment (elrodro83) [inline modules/ws/src/main/java/org/mule/runtime/module/ws/consumer/WSConsumer.java]: static import

File: core/src/main/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategy.java [hunk 4]
```diff
@@ -168,12 +167,7 @@ private void incrementProcessAttemptCountAndRescheduleOrRemoveFromStore(final Se
   }
 
   private Serializable storeEvent(final MuleEvent event) throws ObjectStoreException {
-    Integer configuredAttempts = null;
-    try {
-      configuredAttempts = event.getFlowVariable(PROCESS_ATTEMPT_COUNT_PROPERTY_NAME);
-    } catch (NoSuchElementException nsee) {
-      // Ignore
-    }
+    Integer configuredAttempts = DefaultMuleEvent.getFlowVariableOrNull(PROCESS_ATTEMPT_COUNT_PROPERTY_NAME, event);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategy.java]: static import

**Comment by elrodro83** [comment]
jenkins validate

**Pull request merged**